### PR TITLE
slf4j logger with annotations on mdc context

### PR DESCRIPTION
**Pull request by pierangeloc**
Added an implementation of slf4j logger that puts a defined set of annotations from the LogContext into MDC logging context.

### CODE REVIEW

**Comment by pshemass** [comment]
@pierangeloc looks good to me please add example to documentation and test if possible.

File: slf4j/src/main/scala/zio/logging/slf4j/Slf4jLogger.scala [hunk 1]
```diff
@@ -53,4 +54,40 @@ object Slf4jLogger {
       },
       rootLoggerName = rootLoggerName
     )
+
+  def makeWithAnnotationsAsMdc(
```
Comment (pshemass) [inline slf4j/src/main/scala/zio/logging/slf4j/Slf4jLogger.scala:58]: please add example to documentation and test if possible

Comment (pierangeloc) [reply slf4j/src/main/scala/zio/logging/slf4j/Slf4jLogger.scala:58]: @pshemass sure, will do that 👍

Comment (pshemass) [reply slf4j/src/main/scala/zio/logging/slf4j/Slf4jLogger.scala:58]: @pierangeloc any update on that? do you need help?

Comment (pierangeloc) [reply slf4j/src/main/scala/zio/logging/slf4j/Slf4jLogger.scala:58]: @pshemass sorry I got dragged by other tasks during the week. I'm done with testing, need just to update docs. I should be done with that today/tomorrow

Comment (pierangeloc) [reply slf4j/src/main/scala/zio/logging/slf4j/Slf4jLogger.scala:61]: @pshemass added test and docs, can you have a look?

**Comment by sebver** [comment]
Hi @pierangeloc, I'm implementing something like this myself now and stumbled upon this PR. The code works for me, but I noticed it doesn't take into account the initial value. Is this something we want to do?

Not the best code probably but this way it takes into account:
```
val mdc: Map[String, String] =
    mdcAnnotations
      .asInstanceOf[List[logging.LogAnnotation[Any]]]
      .map(k => k.name -> renderedContext.getOrElse(k.name, k.render(k.initialValue)))
      .toMap
```

**Comment by pshemass** [comment]
@sebver feel free to open PR for your change

**Pull request merged**